cql3/functions/functions.cc

```diff
@@ -572,6 +572,13 @@ functions& instance() {
     return f;
 }
 
+void change_batch::commit() {
+    if (_declared.empty()) {
+        return;
+    }
+    instance()._declared = std::move(_declared);
```

Comment:
Wouldn't it be safer to append to `instance()._declared`? Now we're overriding it.

Reply:
we override because we also support removing functions

Reply:
I was rather thinking about a scenario when commit is invoked twice from different places to first add and then append.

Reply:
that's not supported. technically it should be std::move(batch).commit() but I'm not sure it's worth to declare it like that, for know I didn't do it

Reply:
OK, then I hope nobody will do it the way I imagined it working. The same goes for this comment, I suppose https://github.com/scylladb/scylladb/pull/19598/commits/350c33fb609a25f775819233d90bd1591840ca29#diff-2cc928e139ea64f5257e11df02cd2819944fc93210107633315ee2d0edd05c27R1973-R1974:
```
            // as we don't yield between dropping cache and committing batch
            // change there is no window between cache removal and declaration removal
```
which says what's the current situation, but this may change in the future and we're not protected against it.

Reply:
not sure if we have automated means of protecting from it